statesync/reactor_test.go

```diff
@@ -270,6 +313,157 @@ func TestReactor_SnapshotsRequest(t *testing.T) {
 	}
 }
 
+func TestReactor_LightBlockResponse(t *testing.T) {
+	rts := setup(t, nil, nil, nil, 2)
+
+	var height int64 = 10
+	h := factory.MakeRandomHeader()
+	h.Height = height
+	blockID := factory.MakeBlockIDWithHash(h.Hash())
+	vals, pv := factory.RandValidatorSet(1, 10)
+	vote, err := factory.MakeVote(pv[0], h.ChainID, 0, h.Height, 0, 2,
+		blockID, factory.DefaultTestTime)
+	require.NoError(t, err)
+
+	sh := &types.SignedHeader{
+		Header: h,
+		Commit: &types.Commit{
+			Height:  h.Height,
+			BlockID: blockID,
+			Signatures: []types.CommitSig{
+				vote.CommitSig(),
+			},
+		},
+	}
+
+	lb := &types.LightBlock{
+		SignedHeader: sh,
+		ValidatorSet: vals,
+	}
+
+	require.NoError(t, rts.blockStore.SaveSignedHeader(sh, blockID))
+
+	rts.stateStore.On("LoadValidators", height).Return(vals, nil)
+
+	rts.blockInCh <- p2p.Envelope{
+		From: p2p.NodeID("aa"),
+		Message: &ssproto.LightBlockRequest{
+			Height: 10,
+		},
+	}
+	require.Empty(t, rts.blockPeerErrCh)
+
+	response := <-rts.blockOutCh
+	require.Equal(t, p2p.NodeID("aa"), response.To)
+	res, ok := response.Message.(*ssproto.LightBlockResponse)
+	require.True(t, ok)
+	receivedLB, err := types.LightBlockFromProto(res.LightBlock)
+	require.NoError(t, err)
+	require.Equal(t, lb, receivedLB)
+}
+
+func TestReactor_Dispatcher(t *testing.T) {
+	rts := setup(t, nil, nil, nil, 2)
+	rts.peerUpdateCh <- p2p.PeerUpdate{
+		NodeID: p2p.NodeID("aa"),
+		Status: p2p.PeerStatusUp,
+	}
+	rts.peerUpdateCh <- p2p.PeerUpdate{
+		NodeID: p2p.NodeID("bb"),
+		Status: p2p.PeerStatusUp,
+	}
+
+	closeCh := make(chan struct{})
+	defer close(closeCh)
+
+	chain := buildLightBlockChain(t, 1, 10, time.Now())
+	go handleLightBlockRequests(t, chain, rts.blockOutCh, rts.blockInCh, closeCh, 0)
+
+	dispatcher := rts.reactor.Dispatcher()
+	providers := dispatcher.Providers(factory.DefaultTestChainID, 5*time.Second)
+	require.Len(t, providers, 2)
+
+	wg := sync.WaitGroup{}
+
+	for _, p := range providers {
+		wg.Add(1)
+		go func(t *testing.T, p provider.Provider) {
+			for height := 2; height < 10; height++ {
+				t.Log("height", height)
+				lb, err := p.LightBlock(context.Background(), int64(height))
+				require.NoError(t, err)
+				require.NotNil(t, lb)
+			}
+			wg.Done()
+		}(t, p)
+	}
+	wg.Wait()
```

Comment:
is there a meaningful assertion if it's done

Reply:
I just want to make sure that this completes without locking up. Should I add a timeout or something like that

Reply:
timeouts are great. assertion of some state at the end is good too. 

I think some comment is useful.